pyqtgraph/functions.py

```diff
@@ -1786,7 +1786,10 @@ def downsample(data, n, axis=0, xvals='subsample'):
     sl[axis] = slice(0, nPts*n)
     d1 = data[tuple(sl)]
     d1.shape = tuple(s)
-    d2 = d1.mean(axis+1)
+    if has_nans:
+        d2 = np.nanmean(d1, axis+1)
+    else:
+        d2 = d1.mean(axis+1)
```

Comment:
I'm curious if we should just run `np.nanmean` for all codepaths, or is there a performance penalty for running `nanmean` vs. `mean`?

Reply:
np.nanmean has a significant performance penalty compared to np.mean:

```
>>> timeit.timeit('np.mean(arr)', number=1000, setup='import numpy as np; arr = np.random.random(int(1E6))')
0.24538480000046548
>>> timeit.timeit('np.nanmean(arr)', number=1000, setup='import numpy as np; arr = np.random.random(int(1E6))')
3.1709375999998883
```

Reply:
oh...wow... didn't realize it was _that_ much slower.  Thanks for the post.